packages/core/lib/src/core_widget_factory.dart

```diff
@@ -293,8 +288,7 @@ class WidgetFactory {
   }
 
   /// Returns [ImageProvider].
-  Object imageProvider(ImageSource imgSrc) {
-    if (imgSrc == null) return null;
+  Object? imageProvider(ImageSource imgSrc) {
```

Comment:
Should this return `ImageProvider?` instead of `Object?`?

Reply:
The core WF returns `ImageProvider` but the enhanced one may returns something else for svg images. This is indeed not ideal, I would like to get rid of this if possible.

Reply:
Maybe `imageProvider()` could be removed and instead of passing a provider to `buildImage` it could be passed the `ImageSource` directly? `buildImage` could then call `imageProvider()` internally.

Reply:
Sounds good, I'll send a PR against your branch for this.